src/Redmine/Client/Psr18Client.php

```diff
@@ -24,22 +26,45 @@ final class Psr18Client implements Client
     private ?string $password;
     private ?string $impersonateUser = null;
     private ClientInterface $httpClient;
-    private ServerRequestFactoryInterface $requestFactory;
+    private RequestFactoryInterface $requestFactory;
     private StreamFactoryInterface $streamFactory;
     private ?ResponseInterface $lastResponse = null;
 
     /**
-     * $apikeyOrUsername should be your ApiKey, but it could also be your username.
-     * $password needs to be set if a username is given (not recommended).
+     * @param RequestFactoryInterface|ServerRequestFactoryInterface $requestFactory
+     * @param string $apikeyOrUsername should be your ApiKey, but it could also be your username.
+     * @param ?string $password needs to be set if a username is given (not recommended).
      */
     public function __construct(
         ClientInterface $httpClient,
-        ServerRequestFactoryInterface $requestFactory,
+        $requestFactory,
         StreamFactoryInterface $streamFactory,
         string $url,
         string $apikeyOrUsername,
         string $password = null
     ) {
+        if ($requestFactory instanceof ServerRequestFactoryInterface) {
```

Comment:
We should also check if `$requestFactory` is not an instance of `RequestFactoryInterface` already. `\GuzzleHttp\Psr7\HttpFactory` from the README example implements both interfaces.